mbox/mbox.c

```diff
@@ -1800,8 +1800,8 @@ static enum MxStatus mbox_mbox_check_stats(struct Mailbox *m, uint8_t flags)
                                                       &m->stats_last_checked) > 0)
   {
     bool old_peek = m->peekonly;
-    struct Context *ctx = mx_mbox_open(m, MUTT_QUIET | MUTT_NOSORT | MUTT_PEEK);
-    mx_mbox_close(&ctx);
+    mx_mbox_open(m, MUTT_QUIET | MUTT_NOSORT | MUTT_PEEK);
+    mx_mbox_close(m);
```

Comment:
(your change didn't introduce this problem, but...)

We may need to preserve/restore some flags for code like this.
- `MUTT_QUIET` affects `Mailbox->verbose`
- `MUTT_PEEK` affects `Mailbox->peekonly`